apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/_components/Sidebar/CommitSelector/NewDraftCommitModal/index.tsx

```diff
@@ -0,0 +1,89 @@
+import {
+  Button,
+  CloseTrigger,
+  FormWrapper,
+  Input,
+  Modal,
+  TextArea,
+  useCurrentProject,
+} from '@latitude-data/web-ui'
+import { createDraftCommitAction } from '$/actions/commits/create'
+import { ROUTES } from '$/services/routes'
+import useCommits from '$/stores/commitsStore'
+import { useRouter } from 'next/navigation'
+import { useServerAction } from 'zsa-react'
+
+export default function NewDraftCommitModal({
+  open,
+  setOpen,
+}: {
+  open: boolean
+  setOpen: (open: boolean) => void
+}) {
+  const { data, mutate } = useCommits()
+  const { project } = useCurrentProject()
+  const router = useRouter()
+  // NOTE: `input` in this hook does not work atm because this PR is not merged yet:
+  // https://github.com/IdoPesok/zsa/pull/155
+  // But I tried and it works nicely.
+  const input = { title: '', description: '' } // Fake until the PR is merged
+  const { error, executeFormAction, isPending } = useServerAction(
+    createDraftCommitAction,
+    {
+      persistErrorWhilePending: true,
+      persistDataWhilePending: true,
+      onSuccess: (result) => {
+        const draft = result.data
+        mutate([...data, draft])
+        router.push(
+          ROUTES.projects
+            .detail({ id: project.id })
+            .commits.detail({ uuid: draft.uuid }).root,
+        )
+        setOpen(false)
+      },
+    },
+  )
```

Comment:
put this write method inside commits store and use useFormAction instead

```
const { createDraft } = useCommits
const { action, error, isPending } = useFormAction(createDraft) 
```

Reply:
I find this way more concise. Let's talk offline I'm not convince `useFormAction` is better than this